addon_service/abstract/authorized_account/serializers.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import TYPE_CHECKING
+
+from django.core.exceptions import ValidationError as ModelValidationError
+from rest_framework_json_api import serializers
+
+from addon_service.common.credentials_formats import CredentialsFormats
+from addon_service.osf_models.fields import encrypt_string
+from addon_service.serializer_fields import (
+    CredentialsField,
+    EnumNameMultipleChoiceField,
+)
+from addon_toolkit import AddonCapabilities
+
+
+if TYPE_CHECKING:
+    from addon_service.abstract.authorized_account.models import AuthorizedAccount
+    from addon_service.models import ExternalStorageService
+
+REQUIRED_FIELDS = frozenset(["url", "account_owner", "authorized_operations"])
+
+
+class AuthorizedAccountSerializer(serializers.HyperlinkedModelSerializer):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Check whether subclasses declare all of required fields
+        if not REQUIRED_FIELDS.issubset(set(self.fields.keys())):
+            raise Exception(
+                f"{self.__class__.__name__} requires {self.REQUIRED_FIELDS} to be instantiated"
+            )
+
+        # Check if it's a POST request and remove the field as it's not in our FE spec
+        if "context" in kwargs and kwargs["context"]["request"].method == "POST":
+            self.fields.pop("configured_storage_addons", None)
+
+    display_name = serializers.CharField(
+        allow_blank=True, allow_null=True, required=False, max_length=256
+    )
+    authorized_capabilities = EnumNameMultipleChoiceField(enum_cls=AddonCapabilities)
+    authorized_operation_names = serializers.ListField(
+        child=serializers.CharField(),
+        read_only=True,
+    )
+    auth_url = serializers.CharField(read_only=True)
+    api_base_url = serializers.URLField(allow_blank=True, required=False)
+
+    credentials = CredentialsField(write_only=True, required=False)
+    initiate_oauth = serializers.BooleanField(write_only=True, required=False)
+
+    included_serializers = {
+        "account_owner": "addon_service.serializers.UserReferenceSerializer",
+        "external_storage_service": "addon_service.serializers.ExternalStorageServiceSerializer",
+        "configured_storage_addons": "addon_service.serializers.ConfiguredStorageAddonSerializer",
+        "authorized_operations": "addon_service.serializers.AddonOperationSerializer",
+    }
+
+    @abstractmethod
+    def create_external_service(
+        self,
+        validated_data: dict,
+    ) -> ExternalStorageService:  # TODO: change to ExternalService once implemented
+        ...
+
+    @abstractmethod
+    def create_authorized_account(
+        self,
+        external_service: ExternalStorageService,
+        validated_data: dict,
+    ) -> AuthorizedAccount: ...
```

Comment:
This would be the case if there was a need to create `validated_data` dict, but in current case `validated_data` is an instance of validated data directly passed from serializer

Reply:
I guess what I'm saying is that the `create_authorized_account` logic doesn't/shouldn't need to be aware of the structure of `validated_data`.

So this would be
`def create_authorized_account(self, **kwargs): -> AuthorizedAccount...` and the base `create` function could call `create_authorized_account(**validated_data)`

Then the subclass could define
```
def create_authorized_account(
    self,
    external_storage_service, 
    display_name,
    authorized_capabilities,
    api_base_url,
    **kwargs,
): -> AuthorizedStorageAccount
```
The parameter names would need to align with the serializer field names (since those should align with the model field names), but that doesn't feel problematic to me. Would also potentially remove the need for `get_external_account`.